Public/Src/Tools/VBCSCompilerLogger/VBCSCompilerLogger.cs

```diff
@@ -0,0 +1,191 @@
+    /// </remarks>
+    public class CompilerFileAccessLogger : Logger
+    {
+        private const string CscTaskName = "Csc";
+        private const string VbcTaskName = "Vbc";
+        private const string CscToolName = "csc.exe";
+        private const string VbcToolName = "vbc.exe";
+        private AugmentedManifestReporter m_augmentedReported;
+
+        /// <inheritdoc/>
+        public override void Initialize(IEventSource eventSource)
+        {
+            eventSource.MessageRaised += EventSourceOnMessageRaised;
+            m_augmentedReported = AugmentedManifestReporter.Instance;
+        }
+
+        private void EventSourceOnMessageRaised(object sender, BuildMessageEventArgs e)
+        {
+            if (e is TaskCommandLineEventArgs commandLine)
+            {
+                // We are only interested in CSharp and VisualBasic tasks
+                string language;
+                string arguments;
+                switch (commandLine.TaskName)
+                {
+                    case CscTaskName:
+                        language = LanguageNames.CSharp;
+                        arguments = GetArgumentsFromCommandLine(CscToolName, commandLine.CommandLine);
+                        break;
+                    case VbcTaskName:
+                        language = LanguageNames.VisualBasic;
+                        arguments = GetArgumentsFromCommandLine(VbcToolName, commandLine.CommandLine);
+                        break;
+                    default:
+                        return;
+                }
+
+                // We were able to split the compiler invocation from its arguments. This is the indicator
+                // that something didn't go as expected. Since failing to parse the command line means we
+                // are not catching all inputs/outputs properly, we have no option but to fail the corresponding pip
+                if (arguments == null)
+                {
+                    throw new ArgumentException($"Unexpected tool name in command line. Expected '{CscToolName}' or '{VbcToolName}', but got: {commandLine.CommandLine}");
+                }
+
+                var parsedCommandLine = CompilerUtilities.GetParsedCommandLineArguments(language, arguments, commandLine.ProjectFile);
+                RegisterAccesses(parsedCommandLine);
+            }
+        }
+
+        private string GetArgumentsFromCommandLine(string toolToTrim, string commandLine)
+        {
+            toolToTrim += " ";
+            int index = commandLine.IndexOf(toolToTrim, StringComparison.OrdinalIgnoreCase);
+            
+            if (index == -1)
+            {
+                return null;
+            }
+
+            return commandLine.Substring(index + toolToTrim.Length);
+        }
+
+        private void RegisterAccesses(CommandLineArguments args)
+        {
+            // All inputs

```

Comment:
Why aren't the references considered as inputs?

Reply:
Think it's this one [https://github.com/dotnet/roslyn/blob/a51ee294a07b04cca9d307e50772c87b04f2309d/src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs#L56](https://github.com/dotnet/roslyn/blob/a51ee294a07b04cca9d307e50772c87b04f2309d/src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs#L56)

---
In reply to: [346065602](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346065602) [](ancestors = 346065602)

Reply:
Good catch. I think you found a copy & paste error, observe I'm adding AnalyzerReferences twice, but not ReferencePaths :)

---
In reply to: [346065905](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346065905) [](ancestors = 346065905,346065602)